readthedocs/analytics/utils.py

```diff
@@ -17,15 +17,23 @@
 
 
 def get_client_ip(request):
-    """Gets the real IP based on a request object."""
-    ip_address = request.META.get('REMOTE_ADDR')
+    """
+    Gets the real client's IP address.
 
-    # Get the original IP address (eg. "X-Forwarded-For: client, proxy1, proxy2")
-    x_forwarded_for = request.META.get('HTTP_X_FORWARDED_FOR', '').split(',')[0]
+    It returns the real IP address of the client based on ``HTTP_X_FORWARDED_FOR``
+    header. If ``HTTP_X_FORWARDED_FOR`` is not found, it returns the value of
+    ``REMOTE_ADDR`` header and returns ``None`` if both the headers are not found.
+    """
+    x_forwarded_for = request.META.get('HTTP_X_FORWARDED_FOR', None)
     if x_forwarded_for:
-        ip_address = x_forwarded_for.rsplit(':')[0]
```

Comment:
This logic was removed and also shouldn't be. The X-Forwarded-For header is not exactly a standard but some implementations include a port number. This line strips that port number.

This function is correct and working as-is. Is there a reason to change it? I do appreciate comments which capture the reasoning though.

Reply:
@davidfischer 

> The X-Forwarded-For header is not exactly a standard but some implementations include a port number.

Thank you for this information. I didn't found this while searching and removed this line thinking that it can produce bugs.
I have updated the code.

> Is there a reason to change it?

The only reason was the readability and to improve comments.

Reply:
We should add tests for this also.

Reply:
@ericholscher 
I have added the tests.